pkg/apis/camel/v1/common_types.go

```diff
@@ -93,8 +93,9 @@ type MavenSpec struct {
 	// A JKS formatted keystore is automatically created to store the CA certificate(s),
 	// and configured to be used as a trusted certificate(s) by the Maven commands.
 	// Note that the root CA certificates are also imported into the created keystore.
-	CASecret *corev1.SecretKeySelector `json:"caSecret,omitempty"`
-	Timeout  *metav1.Duration          `json:"timeout,omitempty"`
+	CASecret     *corev1.SecretKeySelector `json:"caSecret,omitempty"`
+	Timeout      *metav1.Duration          `json:"timeout,omitempty"`
+	Repositories []string                  `json:"repositories,omitempty"`
```

Comment:
Would it be possible to use the _structured_ repository definition in the API? The string representation is rather a CLI convenience, and it might be better avoiding leaking it into the API.

Reply:
Ah it's already defined that way in the [IntegrationKitSpec](https://github.com/apache/camel-k/blob/main/pkg/apis/camel/v1/integrationkit_types.go#L35), is that okay ?

Reply:
Ah yes, I think we can deal with the `IntegrationKitSpec` refactoring in #2321. For the `Build` API, we have the opportunity to avoid repeating the mistake of using a serialised string representation for a complex structure.

Reply:
Makes sense, looks like using the [Repostory](https://github.com/apache/camel-k/blob/main/pkg/util/maven/maven_types.go#L29) creates an import cycle due to Maven Log depending on camel v1, will look into it...

Reply:
Yes, the existing `Repository` struct is in a util package, solely used for XML serialisation. It can either be moved to the API module, or we can introduce a similar struct in the API module.

Reply:
Done, I've moved the Repository bits into the API. The alternative would to duplicate them and have a helper method to map from one Repository structure to the other. Thanks !

Reply:
Thanks, and sorry for the extra conflicts 😅.